content/automate/ManageIQ/System/CommonMethods/Utils.class/__methods__/log_object.rb

```diff
@@ -8,6 +8,16 @@ module System
       module CommonMethods
         module Utils
           class LogObject
+            # If you want to log a message and exit without specifying a handle using the global $evm
+            #    ManageIQ::Automate::System::CommonMethods::Utils::LogObject.log_and_exit(msg, code)
+            #
+            # If you want to log a message and exit using a specific handle
+            #    ManageIQ::Automate::System::CommonMethods::Utils::LogObject.log_and_exit(msg, code, handle)
+            #
+            def self.log_and_exit(msg, exit_code, handle = $evm)
+              handle.log('info', "Script ending #{msg} code : #{exit_code}")
+              exit(exit_code)
```

Comment:
@billfitzgerald0120 @mkanoor Is the goal of this method to replace call in our existing model, like:

[ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job/wait_for_completion.rb#L48](https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/wait_for_completion.rb#L48)

and

[ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM/update_provision_status.rb#L9](https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/update_provision_status.rb#L9)

Reply:
@gmcculloug we recently fixed a bug in how MIQ_STOP can be used to a state machine which requires us to use the exit system call. Till now we have refrained from using exits in the newer methods because that causes the spec to exit. This is more a way to ensure we can test methods with exit in them.